lib/features/push_notification/presentation/websocket/web_socket_queue_handler.dart

```diff
@@ -0,0 +1,107 @@
+
+import 'dart:async';
+import 'dart:collection';
+
+import 'package:core/utils/app_logger.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:tmail_ui_user/features/push_notification/presentation/websocket/web_socket_message.dart';
+
+typedef ProcessMessageCallback = Future<void> Function(WebSocketMessage message);
+typedef OnErrorCallback = void Function(dynamic error, StackTrace stackTrace);
+
+class WebSocketQueueHandler {
+  final Queue<WebSocketMessage> _messageQueue = Queue<WebSocketMessage>();
+  final Set<String> _processedMessageIds = {};
+
+  Completer<void>? _processingLock;
+
+  final _queueController = StreamController<WebSocketMessage>.broadcast();
+
+  final ProcessMessageCallback processMessageCallback;
+  final OnErrorCallback? onErrorCallback;
+
+  WebSocketQueueHandler({
```

Comment:
We don't need to do that because we initialize it in the mailbox controller, the mailbox controller is only initialized once when entering the app, on the other hand the mailbox controller always lives with the application even when the mailbox view is hidden.

Reply:
Why we need to keep an global handler in a component-controller? How about to create an instance which manage 2 queues (for Email, Mailbox state)?

Reply:
> Why we need to keep an global handler in a component-controller?

Actually it is not completely global. According to the structure we built. To support both web and mobile we need components to coexist in one screen. Since they only change the view (show and hide) it is not a navigation to a completely new screen. Above all, we still keep the idea of ​​component - global clearly. That is initialization and release only in one component controller.

>  How about to create an instance which manage 2 queues (for Email, Mailbox state)?

IMO, We should not do that. Currently, receiving StateChange from websocket and processing in app are clearly separated for each state of mailbox and email. They are handled on 2 different handlers. Moreover, synchronization of mailbox and email can be done independently and in parallel. We should not let any of the 2 processes wait for each other.
* Note: Synchronizing processing between mailbox and email on the same instant can still be done if we remove 2 handlers for both.